g2core/settings/settings_shapeoko2.h

```diff
@@ -156,70 +157,91 @@
+#define DI1_POLARITY                NORMALLY_CLOSED
 //#define DI1_ACTION                  INPUT_ACTION_STOP
-#define DI1_ACTION                  INPUT_ACTION_NONE
-#define DI1_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI1_ACTION                  INPUT_ACTION_LIMIT
 
 // Xmax
-#define DI2_MODE                    NORMALLY_CLOSED
+#define DI2_ENABLED                 IO_ENABLED
+#define DI2_POLARITY                NORMALLY_CLOSED
 //#define DI2_ACTION                  INPUT_ACTION_STOP
-#define DI2_ACTION                  INPUT_ACTION_NONE
-#define DI2_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI2_ACTION                  INPUT_ACTION_LIMIT
 
 // Ymin
-#define DI3_MODE                    NORMALLY_CLOSED
+#define DI3_ENABLED                 IO_ENABLED
+#define DI3_POLARITY                NORMALLY_CLOSED
 //#define DI3_ACTION                  INPUT_ACTION_STOP
-#define DI3_ACTION                  INPUT_ACTION_NONE
-#define DI3_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI3_ACTION                  INPUT_ACTION_LIMIT
 
 // Ymax
-#define DI4_MODE                    NORMALLY_CLOSED
+#define DI4_ENABLED                 IO_ENABLED
+#define DI4_POLARITY                NORMALLY_CLOSED
 //#define DI4_ACTION                  INPUT_ACTION_STOP
-#define DI4_ACTION                  INPUT_ACTION_NONE
-#define DI4_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI4_ACTION                  INPUT_ACTION_LIMIT
 
 // Zmin
-#define DI5_MODE                    IO_ACTIVE_HIGH   // Z probe
+#define DI5_ENABLED                 IO_ENABLED
+#define DI5_POLARITY                IO_ACTIVE_HIGH   // Z probe
 #define DI5_ACTION                  INPUT_ACTION_NONE
-#define DI5_FUNCTION                INPUT_FUNCTION_NONE
 
 // Zmax
-#define DI6_MODE                    NORMALLY_CLOSED
+#define DI6_ENABLED                 IO_ENABLED
+#define DI6_POLARITY                NORMALLY_CLOSED
 //#define DI6_ACTION                  INPUT_ACTION_STOP
-#define DI6_ACTION                  INPUT_ACTION_NONE
-#define DI6_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI6_ACTION                  INPUT_ACTION_LIMIT
 
 // Amin
-#define DI7_MODE                    IO_MODE_DISABLED
+#define DI7_ENABLED                 IO_DISABLED
+#define DI7_POLARITY                IO_ACTIVE_LOW
 #define DI7_ACTION                  INPUT_ACTION_NONE
-#define DI7_FUNCTION                INPUT_FUNCTION_NONE
 
 // Amax
-#define DI8_MODE                    IO_MODE_DISABLED
+#define DI8_ENABLED                 IO_DISABLED
+#define DI8_POLARITY                IO_ACTIVE_LOW
 #define DI8_ACTION                  INPUT_ACTION_NONE
-#define DI8_FUNCTION                INPUT_FUNCTION_NONE
 
 // Hardware interlock input
-#define DI9_MODE                    IO_MODE_DISABLED
+#define DI9_ENABLED                 IO_DISABLED
+#define DI9_POLARITY                IO_ACTIVE_LOW
 #define DI9_ACTION                  INPUT_ACTION_NONE
-#define DI9_FUNCTION                INPUT_FUNCTION_NONE
```

Comment:
I’m not sure what order you’ll see my comments, so read this part as if it came in first: Thank you!

Your time doing this is appreciated, and my comments are really just picky notes, not flaws.

One other thing to note, for anyone using this, is the `kOutput`N`_PinNumber` are all set to `-1` in `g2core/board/ArduinoDue/gShield-pinout.h`, so they won’t be functional until you assign them numbers of the Due pins.

Also, in the same file, at the top, `OUTPUT1_PWM` etc. are all defined as `0`, but most of them could be set to `1` to enable PWM output functionality. You have to be careful about what pins you choose and what timers they’re on, as you may accidentally entangle to outputs that are on the same timer. That’s a chip-level thing, and not something we can control.

It appears that pins 8-12 are not used by the gShield for other things, and are already PWM pins, so we could possibly use those.

Reply:
I made the changes that you requested and committed them. That's the easy part.

The hard part is I only have 5 days experience with git and github. I don't know how to modify a pull request. Can you point me in the right direction?

Reply:
If you push your commits to the branch that you used to create the original pull request, github will automatically add that to the pending pull request.

Reply:
I did not know that commits on the same branch subsequent to a pull request contribute to the pull request. I committed a change prior to the commit for your requested changes. You may or may not want that commit.

Is there anything that I should do?

Reply:
No worries, that looks great. The only other commit showing is the changes to Atmel Studio files. We’re deprecating support for that, but it doesn’t hurt if someone maintains those files for now. :)

Thank you!